pageserver/src/tenant/layer_map.rs

```diff
@@ -276,17 +277,17 @@ where
     pub(self) fn insert_historic_noflush(&mut self, layer: Arc<L>) -> anyhow::Result<()> {
         let key = historic_layer_coverage::LayerKey::from(&*layer);
         if self.historic.contains(&key) {
-            bail!(
+            error!(
                 "Attempt to insert duplicate layer {} in layer map",
                 layer.short_id()
             );
```

Comment:
Ok, I replaced `Result<()>` with `Result<bool>` to inform caller that duplicate is detected.
But right now it is not handled somehow.
In compaction the only thing we can do when detect duplicate layer is not to send it to S3.
But it is also not correct, because presence of layer  local directory doesn't mean that it was successfully uploaded to S3.